Frogies-Project/Assets/Scripts/StatsSystem/Endurance/EnduranceSystem.cs

```diff
@@ -25,15 +22,35 @@ public void UseEndurance(float amountOfEndurance)
                 _statsController.ProcessModifier(new StatModifier(new Stat(StatType.Endurance, -amountOfEndurance), StatModificatorType.Additive, -1, Time.time)); 
             }
         }
+
+        public void SetMaxEndurance(Stat stat)
+        {
+            var maxEndurance = GetMaxEndurance();
+            if (stat.Type == StatType.MaxEndurance)
+            {
+                _statsController.ProcessModifier(new StatModifier(new Stat(StatType.Endurance,maxEndurance-stat.Value), StatModificatorType.Additive, -1, Time.time));
+            }
```

Comment:
I'm very confused what is the purpose of this, in what cases it will be actually needed?
<img width="867" alt="image" src="https://user-images.githubusercontent.com/50134463/236681217-1dd2626e-5011-4145-810e-db3eb0aa16f4.png">
Also please invert if, it will look cleaner

Reply:
the purpose is to set new max endurance, for example, we have max endurance 100, and we want to have 90. To do that we need to subtract 10
But you're right it need an additional check if new value is less than current value